herbert_core/utilities/classes/@hashable/private/to_hashable_array_.m

```diff
@@ -32,12 +32,16 @@
         if isa(val,'hashable')
             [val,hash,new_hash] = build_hash(val);
             if new_hash
+                obj(j).do_check_combo_arg = false; % if this operation is
+                % part of constructor we do not need to check combo_values.
+                % If may fail.
                 obj(j).(field_name) = val;
+                obj(j).do_check_combo_arg = true;                
             end
             if iscell(hash) % use array of hashes as source for final hash
                 hash = strjoin(hash ,'');
             end
-            tm = typecast(hash,'uint8');
+            tm = uint8(hash); % beware of possibility hash changing type in a future
```

Comment:
it is subject for discussion. If we use 64 bit hash instead of 128bit hash, container search/modifications operation become an order of magnitude faster. 

May be it worth switching to such hash. If hash is uint64, this statement needs to be modified. Tests will show that there is a bug with hashes, but probably in different place than the row 44, so the comment is intended for debugging in this situation.

Reply:
Alex, these are good points and they should be inserted in comments in the code so future developers know what is going on. Please put your explanations for points we raise as comments in the code